Menu only works for task bar

A menu can't be opened if attached to any widget unless any app is open; that is, it seems to depend on taskbar being active.

Demo:
```
menu ("demo") {
  item("yes", Exec "echo yes")
  item("no", Exec "echo no")
}

layout {
  image {
    value = "gtk-about"
    action = Menu "demo"
    css = "* { min-height: 24px; min-width: 24px; padding: 4px; }"
  }
}
```

Try it with absolutely no windows open and it fails, open any window that triggers a taskbar entry and it works.

EDIT: just updated to latest `main`, issue persists.

The main issue is solved. Thanks.

However there are a couple of compiler warnings:

```
[24/47] Compiling C object sfwbar.p/src_misc.c.o
../src/misc.c: In function ‘json_string_by_name’:
../src/misc.c:75:12: warning: return discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   75 |     return json_object_get_string(ptr);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[45/47] Compiling C object sfwbar.p/src_tray.c.o
../src/tray.c: In function ‘tray_invalidate_all’:
../src/tray.c:57:5: warning: implicit declaration of function ‘tray_item_invalidate’; did you mean ‘tray_item_update’? [-Wimplicit-function-declaration]
   57 |     tray_item_invalidate(flow_grid_find_child(iter->data,sni));
      |     ^~~~~~~~~~~~~~~~~~~~
      |     tray_item_update
[47/47] Linking target sfwbar
```

--- a/src/misc.c
+++ b/src/misc.c
@@ -67,7 +67,7 @@ gchar *get_xdg_config_file ( gchar *fname, gchar *extra )
 }
 
 /* get string value from an object within current object */
-gchar *json_string_by_name ( struct json_object *obj, gchar *name )
+const gchar *json_string_by_name ( struct json_object *obj, gchar *name )
 {
   struct json_object *ptr;
 

--- a/src/trayitem.h
+++ b/src/trayitem.h
@@ -38,5 +38,6 @@ GtkWidget *tray_item_new( SniItem *win, GtkWidget *tray );
 SniItem*tray_item_get_sni ( GtkWidget *self );
 void tray_item_update ( GtkWidget *self );
 gint tray_item_compare ( GtkWidget *, GtkWidget *, GtkWidget * );
+void tray_item_invalidate ( GtkWidget *self );
 
 #endif

